### ISSUE TITLE
Does Xgboost do Newton boosting?

### ISSUE THREAD

**Issue by Orbean** [dmlc/xgboost#3227]
In this paper below"", it mentions "Newton boosting" as the boosting algorithm used by XGBoost on page 42, which need to calculate hessian matrix.

"Tree Boosting With XGBoost"
https://brage.bibsys.no/xmlui/bitstream/handle/11250/2433761/16128_FULLTEXT.pdf

As long as I know, calculating hessian matrix is computational demanding.
Is it true that Xgboost perform 2nd order taylor expansion on loss function or just 2nd order differentiation?

**Comment by yordanivanov92** [comment]
I would love to see an answer to this. Also, an overall pseudocode for the xgboost package would be nice, as given for the gbm package.

**Comment by hcho3** [comment]
> calculating hessian matrix is computational demanding

XGBoost does not compute the Hessian matrix. Instead, it computes the second partial derivative of the (element-wise) loss function with respect to the predicted label:
<img width="743" alt="screen shot 2018-05-23 at 12 31 30 pm" src="https://user-images.githubusercontent.com/2532981/40446856-7ae1e5c2-5e85-11e8-8c23-6c3bfbc03956.png">
[Screenshot taken from my master's thesis]

**Comment by hcho3** [comment]
@yordanivanov92 And here's a pseudo-code for gradient boosting:
<img width="612" alt="screen shot 2018-05-23 at 12 38 48 pm" src="https://user-images.githubusercontent.com/2532981/40447116-5b2c437a-5e86-11e8-8623-c417029afb12.png">
[Screenshot taken from my master's thesis]
